### ISSUE TITLE
A deleted note in 404 still appears in our personal history

### ISSUE THREAD

**Issue by rastapopougros** [hackmdio/codimd#1172]
When an owner choose to delete a note, then this note will now return a 404 error. But it still appears in each personal history of any account who visited it, while it's a dead link : deception, bad UX. :( 

This issue doesn't argue about the "should really delete" problem in #766. Whether or not it's really deleted or just marked as deleted, for the UX, there shouldn't have dead links.

**Comment by ccoenen** [comment]
There's currently two places where your history might be stored: It could be in a user's browser in some localstorage object - there we would have no possibility to remove it. But this would only happen for people who don't have an account on the site.

For everyone who has an account, there's a history object stored in the user table. This is not an ideal place to store it, as it is hard to search and modify. Basically, we'd have to go over every single user's history, parse a JSON, delete the entry and than stringify and store it back. I think the better approach would be to make this an actual table in the database, linking user to note via foreign keys.

**Comment by ccoenen** [comment]
I'm marking this as bug, because I totally agree that a deleted document shouldn't remain in a history. Sadly, I don't think it will get resolved very soon.

**Comment by SISheogorath** [comment]
Makes sense. The origin of this problem is bundled to the fact that originally you weren't able to delete a note at all. Which made such a situation close to impossible.

Obviously this requires changes now :+1:

**Comment by rastapopougros** [comment]
Ok, thanks, at least we are agree on the UX problem and it is kept here for memory. :)

Anyway, as long as we will have a real lists of available notes (multiple issues for that, one for all accesible notes #975, and multiple for a particular user notes like #653), we will not use the history to access our notes (or rarely).
